runtime/integration-tests/src/generic/env.rs

```diff
@@ -166,6 +175,58 @@ pub trait Env<T: Runtime>: Default {
 	fn __priv_build_block(&mut self, i: BlockNumber);
 }
 
+pub trait EnvEvmExtension<T: Runtime>: Env<T> {
+	type EvmEnv: EvmEnv<T>;
+
+	/// Allows to mutate the parachain storage state through the closure.
+	fn state_mut<R>(&mut self, f: impl FnOnce(&mut Self::EvmEnv) -> R) -> R;
```

Comment:
Because we have a bunch of these methods, I would call it `parachain_state_mut_with_evm()` or similar. 

Same with `state()`

Reply:
puh. Not sure this does not complicate it more than it needs to be tbh.

Reply:
From a reader's perspective, it was confusing at first for me, what exactly it mutates. Not super strong opinion anyways.

Reply:
I mean, we are always mutating the parachain state, right? In 99% of cases.

Reply:
Also, I think this makes using `trait Env` and `trait EnvEvmExtension` in the same namespace quite annoying, right? Will keep for now. We can revisit later, I anyways wanted to make the `EVM` a global, like Substrate storage. Then we can remove all of that here anyways.

Reply:
Makes sense the latter 👍🏻